cogs/fun.py

```diff
@@ -1,16 +1,272 @@
+
+    @command()
+    async def dog(self, ctx: Context) -> None:
+        """Random doggos just because!"""
+
+        def decide_source() -> str:
+            n = random.random()
+            if n < 0.5:
+                return "https://random.dog/woof"
+            elif n > 0.5:
+                return "https://dog.ceo/api/breeds/image/random"
+
+        async with aiohttp.ClientSession() as session:
+            async with session.get(decide_source(), headers=self.user_agent) as shibe_get:
+                if shibe_get.status == 200:
+                    if shibe_get.host == "random.dog":
+                        shibe_img = await shibe_get.text()
+                        shibe_url = "https://random.dog/" + shibe_img
+                    elif shibe_get.host == "dog.ceo":
+                        shibe_img = await shibe_get.json()
+                        shibe_url = shibe_img["message"]
+
+                    if ".mp4" in shibe_url:
+                        await ctx.send("video: " + shibe_url)
+                    else:
+                        shibe_em = discord.Embed(colour=0x690E8)
+                        shibe_em.set_image(url=shibe_url)
+                        await ctx.send(embed=shibe_em)
+                else:
+                    raise ServiceError(f"could not fetch pupper :( (http {shibe_get.status})")
+
+    @command()
+    async def lizard(self, ctx) -> None:
+        """Shows a random lizard picture"""
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://nekos.life/api/lizard", headers=self.user_agent) as lizr:
+                if lizr.status == 200:
+                    img = await lizr.json()
+                    liz_em = discord.Embed(colour=0x690E8)
+                    liz_em.set_image(url=img["url"])
+                    await ctx.send(embed=liz_em)
+                else:
+                    raise ServiceError(f"something went boom (http {lizr.status})")
+
+    @command()
+    async def why(self, ctx: Context) -> None:
+        """Why _____?"""
+        async with aiohttp.ClientSession() as session:
+            async with session.get("https://nekos.life/api/why", headers=self.user_agent) as why:
+                if why.status == 200:
+                    why_js = await why.json()
+                    why_em = discord.Embed(title=f"{ctx.author.name} wonders...", description=why_js["why"], colour=0x690E8,)
+                    await ctx.send(embed=why_em)
+                else:
+                    raise ServiceError(f"something went boom (http {why.status})")
+
+    @command(aliases=["rhash", "robothash", "rh", "rohash"])
+    async def robohash(self, ctx: Context, *, meme: str) -> None:
+        """text => robot image thing"""
+        try:
+            e = discord.Embed(colour=0x690E8)
+            meme = urllib.parse.quote_plus(meme)
+            e.set_image(url=f"https://robohash.org/{meme}.png")
+            await ctx.send(embed=e)
+        except Exception as e:
```

Comment:
Not many things can go wrong here, except for specific errors

Reply:
idk what error is this